### PR TITLE
Add table_aws_transfer_user

### PR DESCRIPTION
**Pull request by jramosf**
# Integration test logs
<details>
  <summary>Logs</summary>

```
  N/A, Terraform tests will run in turbot/steampipe-plugin-aws.git pipeline
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```sql+postgres
> select server_id, user_name, arn from aws_transfer_user where server_id in (select server_id from aws_transfer_server) and user_name = 'foobar';
+---------------------+-----------+-------------------------------------------------------------------------+
| server_id           | user_name | arn                                                                     |
+---------------------+-----------+-------------------------------------------------------------------------+
| s-xxxxxxxxxxxxxxxxx | foobar    | arn:aws:transfer:us-west-2:123456789901:user/s-xxxxxxxxxxxxxxxxx/foobar |
| s-xxxxxxxxxxxxxxxxx | foobar    | arn:aws:transfer:us-west-2:123456789901:user/s-xxxxxxxxxxxxxxxxx/foobar |
| s-xxxxxxxxxxxxxxxxx | foobar    | arn:aws:transfer:us-west-2:123456789901:user/s-xxxxxxxxxxxxxxxxx/foobar |
+---------------------+-----------+-------------------------------------------------------------------------+
```

```sql+postgres
> select server_id, user_name, home_directory_mappings->0->>'Entry' as entry_home_directory, home_directory_mappings->0->>'Target' as target_home_directory from aws_transfer_user where server_id = 's-xxxxxxxxxxxxxxxxx' limit 2;
+---------------------+-----------+----------------------+------------------------------------------------------------------------+
| server_id           | user_name | entry_home_directory | target_home_directory                                                  |
+---------------------+-----------+----------------------+------------------------------------------------------------------------+
| s-xxxxxxxxxxxxxxxxx | foobarabc | /                    | /transf-bucket2/companies/3edea173-fa04-4177-bdb5-e0951ae6e110/exports |
| s-xxxxxxxxxxxxxxxxx | foobarxyz | /                    | /transf-bucket2/companies/d5f68077-6207-4ef9-baaf-bcbc1e1665c6/exports |
+---------------------+-----------+----------------------+------------------------------------------------------------------------+
```

</details>

Tested OK with live data against AWS APIs. More examples in `README.md`.

### CODE REVIEW

File: docs/tables/aws_transfer_user.md
Comment (misraved) [inline docs/tables/aws_transfer_user.md:1]: @jramosf could you please add the corresponding SQLite example queries?

**Review by misraved — CHANGES_REQUESTED** [review]
Thanks @jramosf for the new table 👍.

Please take a look at the review comments and let us know if you have questions 👍.

File: aws/table_aws_transfer_user.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+			IgnoreConfig: &plugin.IgnoreConfig{
+				ShouldIgnoreErrorFunc: shouldIgnoreErrors([]string{"ResourceNotFoundException"}),
+			},
+			Hydrate: getTransferUser,
+			Tags:    map[string]string{"service": "transfer", "action": "DescribeUser"},
+		},
+		List: &plugin.ListConfig{
+			ParentHydrate: listTransferServers,
+			Hydrate:       listTransferUsers,
+			KeyColumns:    plugin.SingleColumn("server_id"),
```
[the remark this answers is not in the document]
Comment (jramosf) [reply aws/table_aws_transfer_user.go]: Addressed

File: aws/table_aws_transfer_user.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	maxLimit := int32(1000)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			maxLimit = limit
+		}
+	}
+
+	input := &transfer.ListUsersInput{
+		ServerId:   aws.String(d.EqualsQuals["server_id"].GetStringValue()),
```
[the remark this answers is not in the document]
Comment (jramosf) [reply aws/table_aws_transfer_user.go]: Addressed

File: docs/tables/aws_transfer_user.md
Comment (jramosf) [reply docs/tables/aws_transfer_user.md:1]: Done

**Comment by jramosf** [comment]
Hi @misraved , thanks for the review. I've addressed all changes and made the whole thing more efficient (we only use the expensive `DescribeUser` API call when user polls those columns. Resending for review (I've force-pushed my changes).

File: aws/table_aws_transfer_user.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+		return nil, err
+	}
+
+	var userName string
+	var serverID string
+	if h.Item != nil {
+		userName = *h.Item.(TransferUserInfo).UserName
+		serverID = *h.Item.(TransferUserInfo).ServerID
+	} else {
+		userName = d.EqualsQuals["user_name"].GetStringValue()
```
Comment (bigdatasourav) [inline aws/table_aws_transfer_user.go:191]:
```suggestion
		userName = d.EqualsQualString("user_name")
```

File: aws/table_aws_transfer_user.go [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+		serverID = d.EqualsQuals["server_id"].GetStringValue()
```
Comment (bigdatasourav) [inline aws/table_aws_transfer_user.go:192]:
```suggestion
		serverID = d. EqualsQualString("server_id")
```

**Comment by bigdatasourav** [comment]
@jramosf, could you please make the minor changes as suggested above?

**Comment by jramosf** [comment]
> @jramosf, could you please make the minor changes as suggested above?

Done

File: aws/table_aws_transfer_user.go [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+	}
+
+	var userName string
+	var serverID string
+	if h.Item != nil {
+		userName = *h.Item.(TransferUserInfo).UserName
+		serverID = *h.Item.(TransferUserInfo).ServerID
+	} else {
+		userName = d.EqualsQualString("user_name")
+		serverID = d. EqualsQualString("server_id")
```
Comment (bigdatasourav) [inline aws/table_aws_transfer_user.go:192]:
```suggestion
		serverID = d.EqualsQualString("server_id")
```

Comment (bigdatasourav) [reply aws/table_aws_transfer_user.go:192]: @jramosf, could you please fix this?

**Review by bigdatasourav — APPROVED** [review]

**Review by misraved — APPROVED** [review]

**Pull request merged**